pnpm-workspace.yaml

```diff
@@ -3,3 +3,5 @@ packages:
   - 'examples/react/**'
   - 'examples/solid/**'
   - 'examples/vue/**'
+  - '!examples/vue/2*'
```

Comment:
it should be possible to have multiple versions of a dependency for examples. I uncommented these lines and run `pnpm install`, and I could see the correct version inside node_modules

<img width="431" alt="Screenshot 2022-10-29 at 10 00 50" src="https://user-images.githubusercontent.com/1021430/198820742-3410b04f-0062-41ae-9833-f21833e6e1d1.png">

This `package.json` had:

```
  "name": "vue",
  "version": "2.6.14",
```

what would be the next steps for me to try out if that fails?
I can also commit my local changes if you want

Reply:
the `nuxt3` directory is not yet part of `workspaces.json` - is that on purpose?

Reply:
If you go to `cd examples/vue/2.6-basic/` and try `pnpm dev` you will get following error:
```bash
$ pnpm dev

> @tanstack/query-example-vue-2.6-basic@ dev /home/mrmentor/git/query/examples/vue/2.6-basic
> vite

failed to load config from /home/mrmentor/git/query/examples/vue/2.6-basic/vite.config.ts
error when starting dev server:
Error: 

Vue packages version mismatch:

- vue@3.2.37 (/home/mrmentor/git/query/node_modules/.pnpm/vue@3.2.37/node_modules/vue/index.js)
- vue-template-compiler@2.6.14 (/home/mrmentor/git/query/node_modules/.pnpm/vue-template-compiler@2.6.14/node_modules/vue-template-compiler/package.json)
```

As you can see, it picks up `vue@3.2.37` depite `vue@2.6` being specified in package.json
It has to be something with how `pnpm` installs packages and `vite` setup.

Reply:
It's also the same thing for `nuxt` example:

```bash
Nuxi 3.0.0-rc.12                                                                                                                                                                                                            23:21:51
Nuxt 3.0.0-rc.12 with Nitro 0.6.0                                                                                                                                                                                           23:21:51
                                                                                                                                                                                                                            23:21:51
  > Local:    http://localhost:3000/ 
  > Network:  http://172.21.35.143:3000/


 ERROR  Failed to resolve import "ufo" from "virtual:nuxt:/home/mrmentor/git/query/examples/vue/nuxt3/.nuxt/paths.mjs". Does the file exist?                                                                                23:21:53


 ERROR  Failed to resolve entry for package "@tanstack/vue-query". The package may have incorrect main/module/exports specified in its package.json.                                                                        23:21:53


 ERROR  Failed to resolve import "defu" from ".nuxt/app.config.mjs". Does the file exist?                                                                                                                                   23:21:53


 ERROR  Failed to resolve entry for package "@tanstack/vue-query". The package may have incorrect main/module/exports specified in its package.json.
```

If i install packages in that example directly using `yarn` it works fine.

Reply:
hm you are right, but I couldn't find out how to make that better. I'm not sure why the wrong things are being picked up, or if this is a pnpm or a vite issue. Workarounds are to use [shamefully-hoist](https://pnpm.io/npmrc#shamefully-hoist) but we'd have to do that for everything in the monorepro and I really do not want that.

I guess the examples would still work in isolation, e.g. in codesandbox, right?

What we could do is:
- keep them excluded from workspaces
- install manually in the example directories with `npm` or `yarn`
- maybe add a lockfile for those examples as well? But it would lock in the "query" version used so that's not so good I think...

Reply:
There is already a snippet about installation of dependencies in the README.
So if you are ok with excluding those examples from workspaces, like it's done right now i would just merge it.